### PR TITLE
Using file repository

### PR DESCRIPTION
**Pull request by yuriyoleynikov**
Addresses #15

### CODE REVIEW

File: TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                    var item = ReadTodoItem(stream);
+
+                    if (item.Id != itemId)
+                        WriteTodoItem(tempStream, item);
+                    else
+                        deleted = true;
+                }
+
+            if (!deleted)
+                throw new KeyNotFoundException("Item was not found.");
```
Comment (Igorbek) [inline TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs]: temp file would not be deleted if an exception thrown here

File: TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            using (var stream = new BinaryReader(File.Open(GetFileName(userId), FileMode.Open, FileAccess.Read)))
+            using (var tempStream = new BinaryWriter(File.Open(tempFileName, FileMode.Create, FileAccess.Write)))
+                while (NotEndOfStream(stream))
+                {
+                    var _item = ReadTodoItem(stream);
+
+                    if (_item.Id != item.Id)
+                        WriteTodoItem(tempStream, _item);
+                    else
+                        willUpdate = true;
```
Comment (Igorbek) [inline TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs]: you don't have to move writing to the end. move it back

File: TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                }
+
+            if (!willUpdate)
+                throw new KeyNotFoundException("Item was not found.");
```
Comment (Igorbek) [inline TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs]: temp file wouldn't be deleted

Comment (Igorbek) [reply TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs]: BTW, use try-finally to make sure it will be removed always

**Review by Igorbek — CHANGES_REQUESTED** [review]

File: TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs (hunk 1 above)
Comment (yuriyoleynikov) [reply TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs]: Done.

File: TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs (hunk 2 above)
Comment (yuriyoleynikov) [reply TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs]: Done.

File: TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs (hunk 3 above)
Comment (yuriyoleynikov) [reply TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs]: Done.

File: TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+        public IEnumerable<TodoItem> GetTodoListByUser(string userId)
+        {
+            if (userId == null)
+                throw new ArgumentNullException(nameof(userId));
+
+            return GetTodoListByUserInner(userId);
+        }
+
+        public IEnumerable<TodoItem> GetTodoListByUserInner(string userId)
```
Comment (Igorbek) [inline TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs]: should be private

File: TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+            var tempFileName = Path.GetTempFileName();
+            var updated = false;
+
+            try
+            {
+                using (var stream = new BinaryReader(File.Open(GetFileName(userId), FileMode.Open, FileAccess.Read)))
+                using (var tempStream = new BinaryWriter(File.Open(tempFileName, FileMode.Create, FileAccess.Write)))
+                    while (NotEndOfStream(stream))
+                    {
+                        var _item = ReadTodoItem(stream);
```
Comment (Igorbek) [inline TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs]: naming! local vars must be camel-cased without underscore.

**Review by Igorbek — CHANGES_REQUESTED** [review]

File: TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs (hunk 4 above)
Comment (yuriyoleynikov) [reply TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs]: Done.

File: TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs (hunk 5 above)
Comment (yuriyoleynikov) [reply TodoListApp/src/TodoListApp/Models/InFileTodoListRepository.cs]: Done.

**Review by Igorbek — APPROVED** [review]

**Pull request merged**